### PR TITLE
Create a wrapper class for the stream deck

### PR DESCRIPTION
**Pull request by ProbablePrime**
This allows us to create methods and store properties a bit more easily. The constructor takes a HID Device which paves the way for easy testing in which we can just mock the device in a constructor.

### CODE REVIEW

**Comment by Lange** [comment]
I'll review this in a few hours. 👌

File: index.js [hunk 1]
```diff
@@ -18,25 +17,34 @@ if (streamDecks.length > 1) {
...
-streamDeck.on('data', data => {
-	for (let i = 0; i < NUM_KEYS; i++) {
-		const keyPressed = Boolean(data[i + 1]);
-		if (keyPressed !== keyState[i]) {
-			if (keyPressed) {
-				emitter.emit('down', i);
-			} else {
-				emitter.emit('up', i);
+class StreamDeck extends EventEmitter {
+	constructor(streamDeck) {
```
Comment (Lange) [inline index.js]: Do we want to maybe rename this param to `device`? I.e., to instantiate a `StreamDeck` instance, you must provide a `device` to build the class around.

File: index.js [hunk 2]
```diff
@@ -18,25 +17,34 @@ if (streamDecks.length > 1) {
...
+		super();
+		this.streamDeck = streamDeck;
+
+		this.streamDeck.on('data', data => {
```
Comment (Lange) [inline index.js]: Same here. Having a `StreamDeck.streamDeck` property is maybe confusing. `StreamDeck.device` or `StreamDeck.hid` or something might be more intuitive?

File: index.js (hunk 1 above)
Comment (ProbablePrime) [inline index.js]: Yes, fair point

**Comment by Lange** [comment]
LGTM, squash and merge at your leisure 👌

**Comment by Lange** [comment]
Oh actually, this should update README.md as well.

**Comment by Lange** [comment]
Wait, no nevermind, it doesn't change any of the examples. Sorry, long day. Yeah this is good to merge.

**Pull request merged**